backend/Application/Services/UserService.cs

```diff
@@ -222,6 +224,57 @@ public async Task<Response<GetListUsersResponse>> GetListAsync(GetListUsersReque
         return new Response<GetListUsersResponse>(true, response);
     }
 
+    public async Task<Response> IsAbleToDisableUser(Guid id)
+    {
+        var isAnyValidAssignment = await IsAnyValidAssignmentBelongsToUser(id);
+
+        if (isAnyValidAssignment)
+        {
+            return new Response(false, ErrorMessages.CannotDisableUser);
+        }
+
+        return new Response(true, Messages.CanDisableUser);
+    }
+
+    public async Task<Response> DisableUserAsync(DisableUserRequest request)
+    {
+        var userRepository = UnitOfWork.AsyncRepository<User>();
+
+        var user = await userRepository.GetAsync(u => !u.IsDeleted &&
+                                                      u.Id == request.Id &&
+                                                      u.Location == request.Location);
+
+        if (user == null)
+        {
+            return new Response(false, ErrorMessages.NotFound);
+        }
+
+        var isAnyValidAssignment = await IsAnyValidAssignmentBelongsToUser(request.Id);
+
+        if (isAnyValidAssignment)
+        {
+            return new Response(false, ErrorMessages.CannotDisableUser);
+        }
+
+        user.IsDeleted = true;
+
+        await userRepository.UpdateAsync(user);
+        await UnitOfWork.SaveChangesAsync();
+
+        return new Response(true, Messages.ActionSuccess);
+    }
+
+    private async Task<bool> IsAnyValidAssignmentBelongsToUser(Guid userId)
+    {
+        var assignmentRepository = UnitOfWork.AsyncRepository<Assignment>();
+
+        var assignments = await assignmentRepository.ListAsync(a => !a.IsDeleted &&
+                                                                    a.AssignedTo == userId &&
+                                                                    a.State != AssignmentState.Declined);
+
+        return assignments.Any();
```

Comment:
This code   var assignments = await assignmentRepository.ListAsync(a => !a.IsDeleted &&
                                                                    a.AssignedTo == userId &&
                                                                    a.State != AssignmentState.Declined);

        return assignments.Any();

        should be shorter like


    return await assignmentRepository.Any(a => !a.IsDeleted &&
                                                                    a.AssignedTo == userId &&
                                                                    a.State != AssignmentState.Declined);